R/DBI-pool.R

```diff
@@ -24,7 +24,7 @@ setMethod("onDestroy", "DBIConnection", function(object) {
 #' @export
 #' @rdname object
 setMethod("onValidate", "DBIConnection", function(object) {
-  pool <- attr(object, "..metadata", exact = TRUE)$pool
+  pool <- pool_metadata(object)$pool
```

Comment:
This is the change that fixed the underlying problem: now we have a helper that ensures we're always accessing the correct attribute.